bench/utils/__init__.py

```diff
@@ -48,6 +49,37 @@ def is_frappe_app(directory: str) -> bool:
 	return bool(is_frappe_app)
 
 
+def is_valid_frappe_branch(frappe_path:str, frappe_branch:str):
+	""" Check if a branch exists in a repo. Throws InvalidRemoteException if branch is not found
+
+	Uses github's api without auth to query branch.
+	If rate limited by gitapi, requests are sent to github.com
+
+	:param frappe_path: git url
+	:type frappe_path: str
+	:param frappe_branch: branch to check
+	:type frappe_branch: str
+	:raises InvalidRemoteException: branch for this repo doesn't exist
+	"""
+	if "http" in frappe_path and frappe_branch:
+		frappe_path = frappe_path.replace(".git", "")
+		try:
+			owner, repo = frappe_path.split("/")[3], frappe_path.split("/")[4]
+		except IndexError:
+			raise InvalidRemoteException
```

Comment:
Add a message for this? Seems confusing when hit with tracebacks (ref: https://github.com/frappe/bench/issues/1267)

Reply:
Resolved in [cd583f7](https://github.com/frappe/bench/pull/1257/commits/cd583f7822a5aaa29f09cc9ba016b6b23c1d9cc2)